ttyplot.c

```diff
@@ -616,17 +588,21 @@ int main(int argc, char *argv[]) {
             redraw_needed = true;
 
         // Handle signals.
-        if (sigint_pending) {
-            break;
-        }
-        if (sigwinch_pending) {
-            sigwinch_pending = 0;
-            endwin();
-            initscr();
-            erase();
-            refresh();
-            gethw();
-            redraw_needed = true;
+        if (select_ret > 0 && FD_ISSET(signal_read_fd, &read_fds)) {
+            unsigned char signal_number;
+            const int count = read(signal_read_fd, &signal_number, 1);
```

Comment:
Only notice now: Can we have `ssize_t` for a return type once more? it may differ from `int` in size.